packages-exp/auth-exp/src/core/user/additional_user_info.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { IdTokenResponse } from '../../model/id_token';
+import { ProviderId } from '../providers';
+import { AdditionalUserInfo } from '../../model/user';
+
+export function fromIdTokenResponse(
+  idTokenResponse: IdTokenResponse
+): AdditionalUserInfo | null {
+  const { providerId } = idTokenResponse;
+  const profile =
+    typeof idTokenResponse.rawUserInfo === 'string'
+      ? JSON.parse(idTokenResponse.rawUserInfo)
+      : {};
+  const isNewUser =
+    !!idTokenResponse.isNewUser ||
+    idTokenResponse.kind === 'identitytoolkit#SignupNewUserResponse';
```

Comment:
if we rely on the kind, we should extract them to an enum/constant

Reply:
Do we know all the possible values? What are they?